lib/linter.js

```diff
@@ -43,6 +43,8 @@ const MAX_AUTOFIX_PASSES = 10;
  * @property {ASTNode} ast The ESTree AST Program node.
  * @property {Object} services An object containing additional services related
  *      to the parser.
+ * @property {ScopeManager|null} scope The scope of this AST.
+ * @property {Object|null} visitorKeys The visitor keys to traverse this AST.
```

Comment:
`linter` does not have those properties.
This is `typedef` of the result of `parser.parseForESLint` method, so this is public.